tee-worker/omni-executor/aa-contracts/aa-demo-app/src/app/page.tsx

```diff
@@ -0,0 +1,393 @@
+							{currentStep === 1 && (
+								<div>
+									<h2 className="text-xl font-semibold mb-4">
+										Step 1: Connect Your Wallet
+									</h2>
+									<p className="text-gray-600 mb-6">
+										Connect your EVM wallet (MetaMask, etc.) or Solana wallet
+										(Phantom, etc.) to get started.
+									</p>
+									<WalletConnect />
+								</div>
+							)}
+
+							{currentStep >= 2 && (
+								<div>
+									<h2 className="text-xl font-semibold mb-4">
+										Step 2: View Your Omni Account
+									</h2>
+									<p className="text-gray-600 mb-6">
+										Your Omni Account address is pre-calculated using your
+										wallet address and client ID.
+									</p>
+									<AAWalletInfo onAddressCalculated={setAAWalletAddress} />
+								</div>
+							)}
+
+							{currentStep >= 3 && currentStep <= 3 && (
+								<div>
+									<h2 className="text-xl font-semibold mb-4">
+										Step 3: Fund Your Omni Account
+									</h2>
+									<p className="text-gray-600 mb-6">
+										Send some ETH to your Omni Account address to enable
+										Account Abstraction features.
+									</p>
+									<FundingGuide
+										aaWalletAddress={aaWalletAddress}
+										onFundingComplete={() => {
+											console.log("Funding complete callback triggered");
+											fetchEthBalance();
+										}}
+									/>
+								</div>
+							)}
+
+							{currentStep >= 4 && currentStep <= 4 && (
+								<div>
+									<h2 className="text-xl font-semibold mb-4">
+										Step 4: Authorize Root Key
+									</h2>
+									<p className="text-gray-600 mb-6">
+										Set up a root signer that can create sessions and delegate
+										access to your Omni Account.
+									</p>
+									<RootKeyAuthorization
+										aaWalletAddress={aaWalletAddress}
+										isFunded={!!isFunded}
+										onAuthorizationComplete={() => setIsAuthorized(true)}
+									/>
+								</div>
+							)}
+
+							{currentStep >= 5 && (
+								<div>
+									<h2 className="text-xl font-semibold mb-4">
+										Step 5: Start Swapping
+									</h2>
+									<p className="text-gray-600 mb-6">
+										Your Omni Account is ready! Send swap requests to the TEE
+										worker service.
+									</p>
+									{/* TODO */}
+								</div>
+							)}
+
+							{/* Always show completed steps in collapsed form */}
+							<div className="space-y-4">
+								{currentStep > 1 && (
+									<div className="bg-white rounded-lg shadow p-4">
+										<div className="flex items-center justify-between">
+											<div className="flex items-center space-x-3">
+												<div className="w-6 h-6 bg-green-500 rounded-full flex items-center justify-center">
+													<Check className="w-4 h-4 text-white" />
+												</div>
+												<span className="font-medium text-green-700">
+													Wallet Connected
+												</span>
+											</div>
+											<span className="text-sm text-gray-500">
```

Comment:
<img width="966" alt="image" src="https://github.com/user-attachments/assets/bc962522-9f92-4ca7-b9ad-be552d80f837" />
I believe it would be beneficial to add network chain recognition and a switchChain feature here. This would help to understand which chain should switch to. If you have other suitable locations for this feature, feel free to choose as you see fit.